### DISCUSSION TITLE
ceresdb笔记/文档需要补充的地方/小的需求想法

### DISCUSSION THREAD

**Discussion by xxaier** [CeresDB/ceresdb#1018] [Ideas]
## ceresdb笔记/文档需要补充的地方

1. overwite/APPEND模式的介绍 （默认是overwite模式？）
2. PRIMARY KEY(ts) 和 PRIMARY KEY(id, ts) 的区别 （在更新的模式为 overwrite 的情况下 主键保证不可以重复）
PRIMARY KEY 必须要包含 ts 列

## 小的需求想法
1. 支持escape的sql查询

类似 https://docs.rs/tokio-postgres/latest/tokio_postgres/

```
let id = "SOME DATA FROM THE USER";

let rows = client
  .query("SELECT * FROM SomeTable WHERE id = $1", &[&id])
  .await?;
```
有时候参数是来自用户的输入，直接format! 容易被sql注入
比如记录用户的的浏览器头，sec-ch-ua: "Not:A-Brand";v="99", "Chromium";v="112" 这是一个文本字段，还有引号，而且是用户输入的，不escape就没法存入

**Comment by xxaier** [comment]
话说thread_num这个配置目前有意义吗？我看代码都没调用过。
![image](https://github.com/CeresDB/ceresdb/assets/130831741/b046c42f-01fb-4885-987a-fa5a5cb0f7f6)
但是文档中有这个配置。

**Reply by ShiKaiWi** [reply]
This option is not used anymore. I'll remove it.

**Comment by xxaier** [comment]
请问在ceresdb sql中二进制怎么表示？因为我看数据类型支持Varbinary，我在写escape函数

**Reply by ShiKaiWi** [reply]
After #1030, binary data can be encoded as hex string in the sql, an example is:
```
INSERT INTO demo(t, name, value) VALUES(1667374200022, 'ceresdb', x'11')
```

**Comment by ShiKaiWi** [comment]
> 支持escape的sql查询

I'll file an issue to track this feature.

**Comment by tematou** [comment]
依托答辩
